podcasts/Onboarding/Plus/SubscriptionPriceAndOfferView.swift

```diff
@@ -17,64 +17,58 @@ struct SubscriptionPriceAndOfferView: View {
 
     var body: some View {
         VStack(alignment: .leading, spacing: 10) {
-            SubscriptionBadge(tier: product.identifier.subscriptionTier)
-            if let offerDescription = offerDescription(for: product) {
-                Text(offerDescription)
-                    .lineLimit(1)
-                    .fixedSize(horizontal: true, vertical: false)
-                    .foregroundColor(product.identifier.plan == .plus ? Color.black : Color.white)
-                    .padding(EdgeInsets(top: 5, leading: 8, bottom: 5, trailing: 8))
-                    .background(product.identifier.plan == .plus ? Color.plusBackgroundColor2 : Color.patronBackgroundColor)
-                    .textCase(.uppercase)
-                    .font(style: .caption2, weight: .semibold)
-                    .clipShape(.capsule)
-                    .transition(.scale)
+            OfferStack {
+                SubscriptionBadge(tier: product.identifier.subscriptionTier)
+                    .transition(.identity.animation(.none))
+                    .id("sub_badge_" + product.identifier.rawValue)
```

Comment:
This fixes an issue where if you swipe to a different card the badge would animate for some reason when toggling the plan.

By specifying the id here, it forces SwiftUI to treat this as a new instance and then apply the transition which is set to none.